ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/button/ButtonComponentView.kt

```diff
@@ -124,15 +148,17 @@ internal fun ButtonComponentView(
             )
             val totalWidth = stack.width
             val totalHeight = stack.height
+            val stackHeightMinusMargin = totalHeight - marginTopPx - marginBottomPx
+            val stackWidthMinusMargin = totalWidth - marginStartPx - marginEndPx
             layout(
                 width = totalWidth,
                 height = totalHeight,
             ) {
                 stack.placeRelative(x = 0, y = 0)
                 // Center the progress indicator.
                 progress.placeRelative(
-                    x = ((totalWidth / 2f) - (progress.width / 2f)).roundToInt(),
-                    y = ((totalHeight / 2f) - (progress.height / 2f)).roundToInt(),
+                    x = marginStartPx.toInt() + ((stackWidthMinusMargin / 2f) - (progress.width / 2f)).roundToInt(),
+                    y = marginTopPx.toInt() + ((stackHeightMinusMargin / 2f) - (progress.height / 2f)).roundToInt(),
```

Comment:
Should we add some previews for the different scenarios here? Not sure if it's easy to take snapshot tests for the progress indicator though (considering it's animating 😅)...